File: src/domain/item/VasItemEntity.ts
import { ItemEntity, ItemEntityType } from './ItemEntity';

export class VasItemEntity extends ItemEntity {
  constructor(
    public parentId: number,
    itemId: number,
    categoryId: number,
    sellerId: number,
    price: number,
    quantity: number,
  ) {
    super(itemId, categoryId, sellerId, price, quantity);
  }
  getType() {
    return ItemEntityType.VAS;
  }
  getSerializable() {
    return {
      vasItemId: this.itemId,
      vasCategoryId: this.categoryId,
      vasSellerId: this.sellerId,
      price: this.price,
      quantity: this.quantity,
    };
  }
}

File: src/main/java/com/feyz/checkoutcase/app/entity/VasItemEntity.java
package com.feyz.checkoutcase.app.entity;

public class VasItemEntity implements BaseItemEntity {
    @lombok.Getter
    private int itemId;
    @lombok.Getter
    private int vasItemId;
    @lombok.Getter
    private int vasCategoryId;
    @lombok.Getter
    private int vasSellerId;
    @lombok.Getter
    private double price;
    @lombok.Getter
    private int quantity;

    public VasItemEntity(int itemId, int vasItemId, int vasCategoryId, int vasSellerId, double price, int quantity) {
        this.itemId = itemId;
        this.vasItemId = vasItemId;
        this.vasCategoryId = vasCategoryId;
        this.vasSellerId = vasSellerId;
        this.price = price;
        this.quantity = quantity;
    }

    public ItemEntityType getItemEntityType() {
        return ItemEntityType.VasItem;
    }

    @Override
    public double getAmount() {
        return this.quantity * this.price;
    }

    @Override
    public void increaseQuantity(int quantity) {
        this.quantity += quantity;
    }

    @Override
    public int getTotalQuantity() {
        return this.getQuantity();
    }
}
